### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by TannerCypret** [sktime/sktime#7905]
**Describe the bug**
I am using the LSTMFCNClassifier and successfully calling the fit method with attention = false in the constructor. When I set attention=True it breaks when calling the fit method. The error is:

`property 'activity_regularizer' of 'make_attention_lstm.<locals>.AttentionLSTM' object has no setter.`

**To Reproduce**
Added attention=true to the constructor of LSTMClassifier

```python
lstm = LSTMFCNCLassifier(n_epochs=700,batch_size=16,callbacks=[tf.keras.callbacks.EarlyStopping('accuracy',patience=25),tensorboard_callback],attention=True)

lstm.fit(x_train,y_train)
```

**Expected behavior**
This should add an attention layer without crashing

**Additional context**
I am wondering if I am doing anything wrong or if there is anything else I need to do to make attention work. I am not doing anything other setting the flag to True and it crashes. When the flag is false the model works fine and has over 90% accuracy.

**Versions**
<details>
          pip: 24.3.1
       sktime: 0.33.1
      sklearn: 1.5.1
       skbase: 0.8.3
        numpy: 1.26.4
        scipy: 1.14.0
       pandas: 2.2.2
   matplotlib: 3.9.2
       joblib: 1.4.2
        numba: None
  statsmodels: None
     pmdarima: None
statsforecast: None
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: 2.16.2

</details>

**Comment by fkiraly** [comment]
confirmed on current main, python 3.11, windows

**Comment by Ankit-1204** [comment]
> pip: 24.3.1 sktime: 0.33.1 sklearn: 1.5.1 skbase: 0.8.3 numpy: 1.26.4 scipy: 1.14.0 pandas: 2.2.2 matplotlib: 3.9.2 joblib: 1.4.2 numba: None statsmodels: None pmdarima: None statsforecast: None tsfresh: None tslearn: None torch: None tensorflow: 2.16.2

I tried checking this for sktime: 0.36.0, and it seems to work. although it did give this warning

UserWarning: Gradients do not exist for variables ['attention_lstm/attention_lstm_cell/kernel', 'attention_lstm/attention_lstm_cell/recurrent_kernel', 'attention_lstm/attention_lstm_cell/attention_kernel', 'attention_lstm/attention_lstm_cell/bias'] when minimizing the loss. If using `model.compile()`, did you forget to provide a `loss` argument?

which is mentioned in another open issue #6553

**Comment by Ankit-1204** [comment]
> [@Ankit-1204](https://github.com/Ankit-1204) , can you confirm that you do _not_ get the error? I can reproduce it with current `main`, so if you do not get it at `0.36.0` that would be important information. Can you plesae check?

its odd. I again created new editable environment for sktime using venv with following dependencies: 
python: 3.10.6

$ pip list
Package                      Version     Editable project location
---------------------------- ----------- -------------------------
joblib                       1.4.2
keras                        3.9.0
matplotlib                   3.10.1
numpy                        1.26.4
pandas                       2.2.3
scikit-base                  0.8.3
scikit-learn                 1.5.2
scipy                        1.15.2
skbase                       0.4.6
sktime                       0.36.0      D:\Work\sktime\sktime
tensorflow                   2.19.0

and it still worked. only time it failed for me was when i installed sktime==0.33.1 from production.

I considered creating virtual env for different python version, but my conda is not functioning properly and i'm not sure how to create env for different python versions using venv

this may be error from my side, i might be improperly setting up the env.

**Comment by fkiraly** [comment]
This seems to be a duplicate of this: https://github.com/sktime/sktime/issues/6553

**Comment by benHeid** [comment]
I investigated the classifier today. I fear we need to invest a significant amount of time, since a lot of used functions are deprecated and some packages not maintained anymore.

**Comment by fkiraly** [comment]
which packages are we talking about here?

**Comment by benHeid** [comment]
We are talking about tensorflow.keras.backend which is (almost) completely deprecated. 

And the following package which is archived and not longer maintained https://github.com/CyberZHG/keras-self-attention
